core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/BatchToTupleConverter.scala

```diff
@@ -1,13 +1,20 @@
 package edu.uci.ics.amber.engine.architecture.worker.neo
 
+import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerInternalQueue.{
+  DataBatch,
+  DataPayload,
+  DummyPayload,
+  EndPayload
+}
 import edu.uci.ics.amber.engine.common.InputExhausted
 import edu.uci.ics.amber.engine.common.tuple.ITuple
 
 class BatchToTupleConverter(internalQueue: WorkerInternalQueue) {
 
   // save current batch related information
-  private var currentBatch: (Int, Array[ITuple]) = _
-  private var currentTupleIndex = 0
+  private var currentBatch: DataBatch = _
+  private var currentInput = 0
+  private var exhaustedFlag = true
```

Comment:
naming convention: please be consistent in boolean variables, the next line is `allExhausted`
Also please add a comment explain what's the difference of this flag and the `allExhausted` flag below